src/NetscriptFunctions/Infiltration.ts

```diff
@@ -81,5 +129,145 @@ export function NetscriptInfiltration(): InternalAPI<NetscriptInfiltation> {
+          };
+        }
+        // Technically we don't need this check, but it's nicer to have it up-front.
+        // It is a duplicate, so watch out for divergence with the same code in Intro.tsx.
+        const startingSecurityLevel = location.infiltrationData.startingSecurityLevel;
+        const difficulty = calculateDifficulty(startingSecurityLevel);
+        if (difficulty >= MaxDifficultyForInfiltration) {
+          return {
+            success: false,
+            message: "This location is too secure for your current abilities. You cannot infiltrate it.",
+          };
+        }
+        Router.toPage(Page.Infiltration, { location, autoStart: true });
+        return { success: true };
+      })();
+      if (!result.success) {
+        helpers.log(ctx, () => result.message);
+      }
+      return result;
+    },
+    getState: (ctx) => () => {
+      checkAccess(ctx);
+      if (!getPlayerLocation(ctx)) {
+        return null;
+      }
+      return getState();
+    },
+    pressKey: (ctx) => (_key) => {
+      checkAccess(ctx);
+      const key = helpers.string(ctx, "key", _key);
+      const location = getPlayerLocation(ctx);
+      if (!location) {
+        // Perform a no-op if we're not infiltrating. The lack of delay should
+        // help clue people in who are debugging.
+        return Promise.resolve();
+      }
+      // Technically your stats can still change while you are infiltrating,
+      // so this does not reflect the same difficulty as the actual infil.
+      // Like the quirks with hack and grow, this is considered OK.
+      const difficulty = calculateDifficulty(location.infiltrationData.startingSecurityLevel);
+      const hasAugment = Player.hasAugmentation(AugmentationName.WKSharmonizer, true);
+      const noise = difficulty < 1 ? 0 : 10 * difficulty;
+      const delay = (hasAugment ? 0.5 : 1) * (60 + difficulty * 30 + Math.random() * noise);
+      return helpers.netscriptDelay(ctx, delay).then(() => {
+        InfiltrationKeyEvents.emit(key);
+      });
+    },
+    pressSpace: (ctx) => () => {
+      checkAccess(ctx);
+      if (!getPlayerLocation(ctx)) {
+        return;
+      }
+      InfiltrationKeyEvents.emit(" ");
+    },
+    claimRewards: (ctx) => (_faction) => {
+      checkAccess(ctx, 2);
+      const faction = _faction == null ? null : getEnumHelper("FactionName").nsGetMember(ctx, _faction);
+      if (!getPlayerLocation(ctx)) {
+        return 0;
+      }
+      const victory = victoryState.value;
+      if (!victory) {
+        helpers.log(ctx, () => "The current infiltration is not finished yet.");
+        return 0;
+      }
+      const state = stageState.value?.() as { [k: string]: number } | undefined;
+      if (faction == null) {
+        const money = state?.possibleMoneyGain ?? 0;
+        helpers.log(ctx, () => `Sold rewards for ${money === 0 ? "ERROR_UNKNOWN" : formatMoney(money)}`);
```

Comment:
I think Cat mentioned this somewhere else but if market demand is allowed to be 0, then the money reward could also be 0 without it being an error.